### ISSUE TITLE
String collisions are not detected when the strings come from enums

### ISSUE THREAD

**Issue by javcasas** [microsoft/TypeScript#26604]
**TypeScript Version:**  3.1.0-dev.20180822

**Search Terms:** enum string

**Code**

```ts
enum BuildingActions {
    sort = "sort"
}

enum AlarmActions {
    sort = "sort"
}

interface SortBuildingsAction {
    type: BuildingActions.sort,
    buildingCriteria: any
}

interface SortAlarmsAction {
    type: AlarmActions.sort,
    alarmCriteria: any
}

type Action = SortAlarmsAction | SortBuildingsAction;

function reduce(a: Action) {
    switch (a.type) {
        case BuildingActions.sort:
            console.log(a.buildingCriteria)
            break;
        case AlarmActions.sort:
            console.log(a.alarmCriteria);
            break
    }
}
```

**Expected behavior:**
* Type error on lines 22-30: BuildingActions.sort is equals to AlarmActions.sort.
* Dead code detected at lines 26-28.

**Actual behavior:**
Code compiles to Javascript without any warnings. The `case` always enters on the first branch.

**Playground Link:** https://www.typescriptlang.org/play/#src=enum%20BuildingActions%20%7B%0D%0A%20%20%20%20sort%20%3D%20%22sort%22%0D%0A%7D%0D%0A%0D%0Aenum%20AlarmActions%20%7B%0D%0A%20%20%20%20sort%20%3D%20%22sort%22%0D%0A%7D%0D%0A%0D%0Ainterface%20SortBuildingsAction%20%7B%0D%0A%20%20%20%20type%3A%20BuildingActions.sort%2C%0D%0A%20%20%20%20buildingCriteria%3A%20any%0D%0A%7D%0D%0A%0D%0Ainterface%20SortAlarmsAction%20%7B%0D%0A%20%20%20%20type%3A%20AlarmActions.sort%2C%0D%0A%20%20%20%20alarmCriteria%3A%20any%0D%0A%7D%0D%0A%0D%0Atype%20Action%20%3D%20SortAlarmsAction%20%7C%20SortBuildingsAction%3B%0D%0A%0D%0Afunction%20reduce(a%3A%20Action)%20%7B%0D%0A%20%20%20%20switch%20(a.type)%20%7B%0D%0A%20%20%20%20%20%20%20%20case%20BuildingActions.sort%3A%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20console.log(a.buildingCriteria)%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20break%3B%0D%0A%20%20%20%20%20%20%20%20case%20AlarmActions.sort%3A%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20console.log(a.alarmCriteria)%3B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20break%0D%0A%20%20%20%20%7D%0D%0A%7D

**Related Issues:** No related issues found.

**Comment by javcasas** [comment]
@nmain I think I phrased it wrong. The previous version of the code was:
```
interface SortBuildingsAction {
    type: "sort",
    buildingCriteria: any
}

interface SortAlarmsAction {
    type: "sort",
    alarmCriteria: any
}

type Action = SortAlarmsAction | SortBuildingsAction;

function reduce(a: Action) {
    switch (a.type) {
        case "sort":
            console.log(a.buildingCriteria)
            break;
        case "sort":
            console.log(a.alarmCriteria);
            break
    }
}
```
and TypeScript notes that something is wrong with the types there. But if I go and make the strings enums, TypeScript is no longer able to spot the issue.

**Comment by Kingwl** [comment]
@RyanCavanaugh 
Technically, different enums should not be equal.
would we disable the symbol check for the enum literal?
Or limited to StringLiteral enum?

**Comment by RyanCavanaugh** [comment]
@Kingwl This should just be a specific logic in detecting duplicate `case` labels. Not sure if that answers your question

**Comment by RyanCavanaugh** [comment]
We should track this at #53604, since even the case of bare string literals isn't checked today (I confused a lint rule for TS's own behavior).
